plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.ts

```diff
@@ -212,28 +212,37 @@ export class DefaultCatalogProcessingEngine implements CatalogProcessingEngine {
                 locationKey,
                 refreshKeys: result.refreshKeys,
               });
-            oldRelationSources = new Set(
-              previous.relations.map(r => r.source_entity_ref),
+            oldRelationSources = new Map(
+              previous.relations.map(r => [
+                `${r.source_entity_ref}:${r.target_entity_ref}:${r.type}`,
```

Comment:
I wanted to have less duplicated logic here, but nothing felt right with the type definitions for `DbRelationsRow` being only in `src/database`. Didn't want to cross-pollinate and couple these internal logical groups between `src/processing` and `src/database`, so just threw this here like this.

Happy to discuss ways to DRY out code here

Reply:
You don't technically need the target here or below, right? Or am I misunderstanding

Reply:
You know that is extremely true - I'll update to remove it!

Reply:
Removed in https://github.com/backstage/backstage/pull/17190/commits/f25bc5cc030d33cb3946d78ed485e45b44e54b12